### PR TITLE
add gnu readline

### PR DESCRIPTION
**Pull request by croth1**
Uses pkg-config to link against conda's ncurses. Let's see how it goes.

### CODE REVIEW

File: recipes/readline/build.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+./configure --prefix=$PREFIX
```
Comment (scopatz) [inline recipes/readline/build.sh]: Maybe add an `|| { cat config.log; exit 1; }` at the end of this line, in case configuring fails.

**Comment by jakirkham** [comment]
Also, I would recommend taking a look at the recipe that @jjhelmus made for this too ( https://github.com/acpd/acpd/tree/osx_build/recipes/readline ). In fact, we can take it verbatim if needed. Though I'll let you play with this a bit.

Please let me know if you have questions. Will try to check back in early next week.

**Comment by msarahan** [comment]
LGTM - please chime in @jjhelmus, since you already put some effort into this, too.

**Comment by croth1** [comment]
```
Package: readline-6.3-0
no source
+ ./configure --prefix=/opt/conda/envs/_build
/conda-recipes/readline/build.sh: line 3: ./configure: No such file or directory
+ cat config.log
cat: config.log: No such file or directory
Command failed: /bin/bash -x -e /conda-recipes/readline/build.sh
```
Any idea what's the problem here?

**Comment by jakirkham** [comment]
Yes, this issue ( https://github.com/conda-forge/staged-recipes/issues/613 ). It's already been identified and a fix has been made. Is it already deployed, @pelson?

**Comment by jjhelmus** [comment]
LGTM. I took my readline recipe from [anaconda-recipes](https://github.com/ContinuumIO/anaconda-recipes), this one looks more complete.

File: recipes/readline/build.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+./configure --prefix=$PREFIX || { cat config.log; exit 1; }
+make SHLIB_LIBS="$(pkg-config --libs ncurses)"
```
Comment (jakirkham) [inline recipes/readline/build.sh]: `make check`?

Comment (croth1) [inline recipes/readline/build.sh]:
```
+ make check
make: Nothing to be done for `check'.
```
Does not seem to provide any tests.

Comment (jakirkham) [inline recipes/readline/build.sh]: That seems to be correct. Worrisome, but I guess there is nothing we can do about it.

**Comment by jakirkham** [comment]
Thanks for adding the patches btw, @croth1.

LGTM too.

**Comment by jakirkham** [comment]
Thanks everyone for reviewing. Also, thanks @croth1 for getting this going.

**Pull request merged**